include/experimental/__p2642_bits/layout_padded.hpp

```diff
@@ -0,0 +1,775 @@
+      // value that is the least multiple of `_PaddingStride` greater than or equal to `_Extents::static_extents(0)`
+      if constexpr (_Extents::static_extent(_ExtentToPadIdx) == dynamic_extent)
+      {
+        const auto __s = __find_aligned_offset(_PaddingStride, __extents.extent(_ExtentToPadIdx));
+        return __subs_type::__construct(__extents, extents<typename _Extents::index_type, _ActualPaddingStride>(__s));
+      } else {
+        return __subs_type::__construct(__extents, extents<typename _Extents::index_type, _ActualPaddingStride>{});
+      }
+    }
+  }
+
+  template <size_t _PaddingStride, class _Size>
+  MDSPAN_INLINE_FUNCTION static constexpr auto
+  __construct(const _Extents &__extents, _Size __padding_value)
+  {
+    const auto __s = __find_aligned_offset(__padding_value, __extents.extent(_ExtentToPadIdx));
+
+    return __subs_type::__construct(__extents, extents<typename _Extents::index_type, _ActualPaddingStride>(__s));
+  }
+
+  template <size_t _PaddingStride, class _RankType, class _OtherMapping>
+  MDSPAN_INLINE_FUNCTION static constexpr auto
+  __construct_other(const _Extents &__extents, _RankType __padding_stride_idx, const _OtherMapping &__other_mapping)
+  {
+    return __subs_type::__construct(__extents, extents<typename _Extents::index_type, _ActualPaddingStride>(__other_mapping.stride(__padding_stride_idx)));
+  }
+};
+
+template<size_t _ExtentToPadIdx, class _Extents, size_t _ActualPaddingStride>
+struct __inner_extents_type<_ExtentToPadIdx, _Extents, _ActualPaddingStride, enable_if_t<_Extents::rank() <= size_t(1)>>
+{
+  using __type = _Extents;
+
+  template<size_t _PaddingStride>
+  MDSPAN_INLINE_FUNCTION
+  static constexpr const _Extents &
+  __construct(const _Extents &__extents)
+  {
+    return __extents;
+  }
+
+  template <size_t _PaddingStride, class _Size>
+  MDSPAN_INLINE_FUNCTION static constexpr auto
+  __construct(const _Extents &__extents, _Size)
+  {
+    return __extents;
+  }
+
+  template <size_t _PaddingStride, class _RankType, class _OtherMapping>
+  MDSPAN_INLINE_FUNCTION static constexpr auto
+  __construct_other(const _Extents &__extents, _RankType, const _OtherMapping &)
+  {
+    return __extents;
+  }
+};
+
+template <size_t _ExtentToPadIdx, class _Extents, typename _Enabled = void>
+struct __unpadded_extent_type_impl
+{
+  using __type = extents<typename _Extents::index_type, _Extents::static_extent(_ExtentToPadIdx)>;
+
+  MDSPAN_INLINE_FUNCTION
+  static constexpr auto
+  __construct(const _Extents &__extents)
+  {
+    return __type(__extents.extent(_ExtentToPadIdx));
+  }
+};
+
+template <size_t _ExtentToPadIdx, class _Extents>
+struct __unpadded_extent_type_impl<_ExtentToPadIdx, _Extents, enable_if_t<(_Extents::rank() == 0)>>
+{
+  using __type = extents<typename _Extents::index_type>;
+
+  MDSPAN_INLINE_FUNCTION
+  static constexpr auto
+  __construct([[maybe_unused]] const _Extents &__extents)
+  {
+    return __type{};
+  }
+};
+}
```

Comment:
Optionally, would you consider replacing this (C++98-y) struct specialization approach with function overloads?

```suggestion
template <size_t _ExtentToPadIdx, class _IndexType, size_t ... _Extents>
MDSPAN_INLINE_FUNCTION
static constexpr auto
__construct_unpadded_extent(const extents<_IndexType, _Extents...>& __extents, integral_constant<size_t, _ExtentToPadIdx>)
{
  using __input_type = extents<_IndexType, _Extents...>;
  using __return_type = extents<_IndexType, __input_type::static_extent(_ExtentToPadIdx)>;
  return __return_type(__extents.extent(_ExtentToPadIdx));
}

template <size_t _ExtentToPadIdx, class _IndexType>
MDSPAN_INLINE_FUNCTION
static constexpr auto
__construct_unpadded_extent(const extents<_IndexType>& __extents, integral_constant<size_t, _ExtentToPadIdx>)
{
  return extents<_IndexType>{};
}
```